hpx/runtime/threads/thread_helpers.hpp

```diff
@@ -770,15 +786,64 @@ namespace hpx { namespace applier
     /// \note All other arguments are equivalent to those of the function
     ///       \a threads#register_thread_plain
     ///
-    HPX_API_EXPORT threads::thread_id_type register_thread(
-        util::unique_function_nonser<void(threads::thread_state_ex_enum)> && func,
+    namespace detail
+    {
+        template <typename F>
+        struct thread_function
+        {
+            typename std::decay<F>::type f;
+
+            inline threads::thread_result_type operator()(threads::thread_arg_type)
+            {
+                // execute the actual thread function
+                std::forward<F>(f)(threads::wait_signaled);
+
+                // Verify that there are no more registered locks for this
+                // OS-thread. This will throw if there are still any locks
+                // held.
+                util::force_error_on_lock();
+
+                return threads::thread_result_type(threads::terminated, nullptr);
+            }
+        };
+
+        template <typename F>
+        struct thread_function_nullary
+        {
+            typename std::decay<F>::type f;
+
+            inline threads::thread_result_type operator()(threads::thread_arg_type)
+            {
+                // execute the actual thread function
+                f();
```

Comment:
Shouldn't this be forwarded as well (same as https://github.com/STEllAR-GROUP/hpx/pull/3114/files#diff-f5e3956f6930a2fde1e87c0917f65d6bR799)?

Reply:
The forward has no effect here. It should rather be removed on the other spot. Good catch.